packages/client/src/views/SysAdmin/Team/user/UserList.tsx

```diff
@@ -404,6 +468,16 @@ function UserListComponent(props: IProps) {
     }
   }
 
+  const getUserName = (user: GQLUser) => {
+    const userName =
+      (user &&
+        user.name &&
+        ((createNamesMap(user.name as GQLHumanName[])[intl.locale] as string) ||
+          (createNamesMap(user.name as GQLHumanName[])[LANG_EN] as string))) ||
+      ''
+    return userName
+  }
+
```

Comment:
https://github.com/opencrvs/opencrvs-core/blob/4d5fcbd4d4a3809250db0d3e9e9c2b3d8741a424/packages/client/src/utils/userUtils.ts#L171
Could we use this function here? I feel like user name generation should be a generic function somewhere in username utilities 🤔

Reply:
Types are not the same on soignature. That's why make a different one. But yeah we should use a generic function for getUserName